File: Samples/Silverlight_RiaServices_Hierarchical_CS/Silverlight_RiaServices.Web/Validators/Territories.Validation.cs
/*===============================================================================
                    EntitySpaces Studio by EntitySpaces, LLC
             Persistence Layer and Business Objects for Microsoft .NET
             EntitySpaces(TM) is a legal trademark of EntitySpaces, LLC
                          http://www.entityspaces.net
===============================================================================
EntitySpaces Version : 2012.1.0930.0
EntitySpaces Driver  : SQL
Date Generated       : 9/23/2012 6:16:17 PM
===============================================================================
*/

using System;
using System.ComponentModel.DataAnnotations;

namespace BusinessObjects
{
    [MetadataType(typeof(TerritoriesValidation))]
    public partial class Territories
    {

    }
	
    internal class TerritoriesValidation
    {

		[KeyAttribute()]
		[StringLength(20)]
		[Required]
		public System.String TerritoryID { get; set; }

		[StringLength(50)]
		[Required]
		public System.String TerritoryDescription { get; set; }

		[Required]
		public System.Int32? RegionID { get; set; }
    }
}

File: Samples/Silverlight_RiaServices_Hierarchical_VB/Silverlight_RiaServices.Web/Validators/Territories.Validation.vb
'===============================================================================
'                   EntitySpaces Studio by EntitySpaces, LLC
'            Persistence Layer and Business Objects for Microsoft .NET
'            EntitySpaces(TM) is a legal trademark of EntitySpaces, LLC
'                         http://www.entityspaces.net
'===============================================================================
' EntitySpaces Version : 2012.1.0930.0
' EntitySpaces Driver  : SQL
' Date Generated       : 9/23/2012 6:16:25 PM
'===============================================================================

Imports System
Imports System.ComponentModel.DataAnnotations

Namespace BusinessObjects

	<MetadataType(GetType(TerritoriesValidation))> _
	Public Partial Class Territories

	End Class

    Friend Class TerritoriesValidation
    

		<KeyAttribute()> _
		<StringLength(20)> _
		<Required> _
		Public Property TerritoryID As System.String

		<StringLength(50)> _
		<Required> _
		Public Property TerritoryDescription As System.String

		<Required> _
		Public Property RegionID As Nullable(Of System.Int32)
    
	End Class
End Namespace